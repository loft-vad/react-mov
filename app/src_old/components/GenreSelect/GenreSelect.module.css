.wrapper {
  display: flex;
  border-bottom: 1px solid #080707;
}
.wrapper input {
  background: rgba(50, 50, 50, 0.8);
  mix-blend-mode: normal;
  opacity: 0.7;
  border-radius: 4px;
  width: 100%;
  border: none;
  padding: 17px 19px 16px;
  font-style: normal;
  font-weight: 400;
  mix-blend-mode: normal;
  opacity: 0.3;
}
.wrapper ul {
  display: flex;
  gap: 30px;
  border-bottom: 2px solid #424242;
  margin-bottom: 1px;
  width: 100%;
}
.wrapper ul li {
  border-bottom: 3px solid transparent;
  margin-bottom: -3px;
}
.wrapper ul li.active {
  border-color: #F65261;
}
.wrapper ul li button {
  color: #fff;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
  text-transform: uppercase;
  padding: 20px 0;
}

/*# sourceMappingURL=GenreSelect.module.css.map */
