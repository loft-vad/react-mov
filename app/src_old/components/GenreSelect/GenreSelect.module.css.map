{"version":3,"sourceRoot":"","sources":["GenreSelect.module.scss"],"names":[],"mappings":"AAAA;EACE;EACA;;AAEA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EAEA;EACA;EAEA;EACA;;AAGF;EACE;EACA;EACA;EACA;EACA;;AAEA;EACE;EACA;;AAEA;EACE;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA","file":"GenreSelect.module.css"}