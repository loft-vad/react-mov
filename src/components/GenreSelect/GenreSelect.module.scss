.wrapper {
  display: flex;
  border-bottom: 1px solid #080707;

  input {
    background: rgba(50, 50, 50, 0.8);
    mix-blend-mode: normal;
    opacity: 0.7;
    border-radius: 4px;
    width: 100%;
    border: none;
    padding: 17px 19px 16px;

    font-style: normal;
    font-weight: 400;

    mix-blend-mode: normal;
    opacity: 0.3;
  }

  ul {
    display: flex;
    gap: 30px;
    border-bottom: 2px solid #424242;
    margin-bottom: 1px;
    width: 100%;

    li {
      border-bottom: 3px solid transparent;
      margin-bottom: -3px;

      &.active {
        border-color: #F65261;
      }

      button {
        color: #fff;
        font-weight: 500;
        font-size: 16px;
        line-height: 20px;
        text-transform: uppercase;
        padding: 20px 0;
      }
    }
  }
}