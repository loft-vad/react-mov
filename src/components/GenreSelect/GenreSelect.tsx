import React from 'react';
import styles from './GenreSelect.module.scss';

export interface genre {
  id: number;
  name: string;
}

export interface GenreProps {
  genres: genre[];
  activeItem?: string;
  onSelect: (name: genre['name']) => void;
}

const GenreSelect: React.FC<GenreProps> = ({
  genres = [],
  activeItem = genres[0].name,
  onSelect,
}) => {
  return (
    <div className={styles.wrapper}>
      <ul>
        {genres.map(({ id, name }) => (
          <li key={id} className={activeItem === name ? styles.active : ''}>
            <button onClick={() => onSelect(name)}>{name}</button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default GenreSelect;
