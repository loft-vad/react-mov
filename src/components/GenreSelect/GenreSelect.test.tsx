import React from "react";

import GenreSelect, { genre } from "./GenreSelect";
import { fireEvent, render } from "@testing-library/react";

const genres: genre[] = [
  { id: 1, name: "All" },
  { id: 2, name: "Documentary" },
  { id: 3, name: "Comedy" },
  { id: 4, name: "Horror" },
  { id: 5, name: "Crime" },
];

describe("GenreSelect component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<GenreSelect genres={genres} onSelect={() => {}} />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
  it("renders all genres passed in props", () => {
    const wrapper = render(<GenreSelect genres={genres} onSelect={() => {}} />);
    const renderedItems = wrapper.getAllByTestId("genre").map((item) => item.textContent);
    const inputItems = genres.map((item) => item.name);
    expect(renderedItems).toEqual(inputItems);
  });
  it("highlights a selected genre passed in props", () => {
    const activeItem = genres[3].name;
    const wrapper = render(
      <GenreSelect genres={genres} onSelect={() => {}} activeItem={activeItem} />,
    );
    const activeItemElement = wrapper.container.querySelector(".active");
    expect(activeItemElement?.textContent).toEqual(activeItem);
  });
  it('after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments', () => {
    const callback = jest.fn();
    const wrapper = render(<GenreSelect genres={genres} onSelect={callback} />);
    const button = wrapper.container.querySelector("button");
    fireEvent.click(button!);
    expect(callback).toHaveBeenCalledWith(genres[0].name);
  });
});
