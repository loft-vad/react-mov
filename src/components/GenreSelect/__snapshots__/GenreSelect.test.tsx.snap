// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelect component should match snapshot 1`] = `
<div
  class="wrapper"
>
  <ul>
    <li
      class="active"
    >
      <button
        data-testid="genre"
      >
        All
      </button>
    </li>
    <li
      class=""
    >
      <button
        data-testid="genre"
      >
        Documentary
      </button>
    </li>
    <li
      class=""
    >
      <button
        data-testid="genre"
      >
        Comedy
      </button>
    </li>
    <li
      class=""
    >
      <button
        data-testid="genre"
      >
        Horror
      </button>
    </li>
    <li
      class=""
    >
      <button
        data-testid="genre"
      >
        Crime
      </button>
    </li>
  </ul>
</div>
`;
